synapse/rest/client/v2_alpha/sync.py

```diff
@@ -131,7 +131,18 @@ def on_GET(self, request):
                 self.filtering.check_valid_filter(filter_object)
                 filter = FilterCollection(filter_object)
             else:
-                filter = yield self.filtering.get_user_filter(user.localpart, filter_id)
+                try:
+                    filter = yield self.filtering.get_user_filter(
+                        user.localpart, filter_id
+                    )
+                except StoreError as err:
+                    if err.code == 404 and err.errcode == Codes.UNKNOWN:
+                        # fix up the description and errcode to be more useful
+                        raise SynapseError(
+                            404, "No such filter", errcode=Codes.NOT_FOUND
```

Comment:
it's not obvious to me that `/sync` with an unknown filter id should give a 404.